coordinator/src/main/scala/filodb.coordinator/ShardAssignmentStrategy.scala

```diff
@@ -31,8 +33,67 @@ trait ShardAssignmentStrategy {
                         resources: DatasetResourceSpec,
                         mapper: ShardMapper): Int
 
+}
+
+/**
+ * Uses the trailing number in the host name to determine the shards that can be mapped to the host
+ * the implementation falls back to DefaultShardAssignmentStrategy if assignment cannot be determined
+ *
+ * The implementation assumes the number of shards is a multiple of number of nodes and there are no
+ * uneven assignments of shards to nodes
+ */
+object K8sStatefulSetShardAssignmentStrategy extends ShardAssignmentStrategy with StrictLogging {
+
+  private val pat = "-\\d+$".r
+
+  private def getOrdinalFromActorRef(coord: ActorRef): Option[(String, Int)] = {
+    // if hostname is None from coordinator actor path, then its a local actor
+    // If the host name does not contain an ordinal at the end (e.g filodb-host-0, filodb-host-10), it will match None
+    coord.path.address.host
+      .map(host => InetAddress.getByName(host).getHostName)
+      .orElse(Some(InetAddress.getLocalHost.getHostName))
+      // TODO: Is there a better way? This returns from the FQDN the hostname part only
+      .map(name => if (name.contains(".")) name.substring(0, name.indexOf('.')) else name)
+      .flatMap(hostName => pat.findFirstIn(hostName).map(ordinal => (hostName, -Integer.parseInt(ordinal))))
   }
 
+  override def shardAssignments(coord: ActorRef,
+                                dataset: DatasetRef,
+                                resources: DatasetResourceSpec,
+                                mapper: ShardMapper): Seq[Int] =
+      getOrdinalFromActorRef(coord) match {
+        case Some((hostName, ordinal)) =>
+          val numShardsPerHost = resources.numShards / resources.minNumNodes
```

Comment:
This needs tweaking. If there are 10 shards, and minNodes = 3, one shard will never be assigned.

Ceil division is an alternative, but it would result in 4,4,2 as opposed to the ideal 4,3,3.

Reply:
I have called this out in the comments of the class

```
  The implementation assumes the number of shards is a multiple of number of nodes and there are no
  uneven assignments of shards to nodes
  ```

IMO its a misconfiguration to have unequal shards on each pod making imbalanced use of attached persistent volume if any in any Kubernetes stateful set. I will not complicate the setup rather have a warning and fallback to default strategy in case ```numShards % minNodes```  is not equal to 0